### PR TITLE
Adding A Complete Guide to Standard C++ Algorithms

### PR DESCRIPTION
**Pull request by HappyCerberus**
## What does this PR do?
Adding my free C++ book about standard algorithms.

### Description

A guidebook to standard C++ algorithms. Covers all algorithms from the standard library.

### Why is this valuable (or not)?

I'm the author so I'm biased. The book is a good introduction to standard algorithms, demonstrating behaviours through code examples and noting relevant language features and caveats.

### How do we know it's really free?

I'm the sole author. Licensed CC-BY-SA-NC.

### For book lists, is it a book? For course lists, is it a course? etc.
It is a book.

## Checklist:
- [X] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md)
- [X] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [X] Include author(s) and platform where appropriate.
- [X] Put lists in alphabetical order, correct spacing.
- [X] Add needed indications (PDF, access notes, under construction)

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

**Comment by HappyCerberus** [comment]
Just to clarify, do you want me to link directly to the pdf in the releases?

But then I will need to update the link whenever I make a change.

**Comment by HappyCerberus** [comment]
Or should I change the link to point to releases?
https://github.com/HappyCerberus/book-cpp-algorithms/releases

**Comment by eshellman** [comment]
Aha! I think that because you have marked your releases as "pre-release", the GitHub UI doesn't offer the "Latest" link or the usual releases UI. That's why I didn't find the PDF.

https://docs.github.com/en/repositories/releasing-projects-on-github/linking-to-releases

The link you have is fine, but your README page ought to link to the latest release.

**Comment by HappyCerberus** [comment]
The README now links directly to the latest PDF.

File: books/free-programming-books-langs.md [hunk 1]
```diff
@@ -492,6 +492,7 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
 
 ### <a id="cpp"></a>C++
 
+* [A Complete Guide to Standard C++ Algorithms](https://github.com/HappyCerberus/book-cpp-algorithms) - Šimon Tóth (PDF with LaTeX source) (:construction: *in process*)
```
Comment (eshellman) [inline books/free-programming-books-langs.md:495]:
```suggestion
* [A Complete Guide to Standard C++ Algorithms](https://github.com/HappyCerberus/book-cpp-algorithms) - Šimon Tóth (PDF, LaTeX) (:construction: *in process*)
```

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!
Thanks for making you book free! If it has an Open License, consider adding it to Unglue.it and Archive.org.

**Pull request merged**